communitiesuk/funding-service-design-fund-application-builder
# FS-4652: implement fund and round update features

Edit fund and round config: https://dluhcdigital.atlassian.net/browse/FS-4652


### Change description
- New Routes: Added /fund/<fund_id> and /round/<round_id> routes to enable updating fund and round details via forms.
- Frontend Enhancements: Introduced “Edit” buttons in the fund_config template for easier access to update forms, used fund and round templates to handle create and update.
- JSON Column Mutability: Addressed SQLAlchemy’s default immutability of JSON columns by using MutableDict.as_mutable to ensure changes are tracked and persisted.
- Form and Database Synchronization: Fixed discrepancies between form inputs and database fields, particularly for minute fields.
- Testing Improvements: 
   - Created test_routes.py to cover the new routes.
   - Developed helpers.py to include utility functions for the new tests.



### How to test
- Add a new fund using "Add a Fund" then in "Manage Application Configuration" select that fund and click on "Edit Fund" button after updating the Fund verify again coming to the same place if it did actually change
- Same goes for updating and verifying the round

### Screenshots of UI changes (if applicable)

Fund Update:
<img width="1398" alt="Screenshot 2024-09-27 at 04 13 37" src="https://github.com/user-attachments/assets/116e870e-0667-4f0e-b2b8-1478cc275e1b">
 
 
<img width="1305" alt="Screenshot 2024-09-27 at 04 14 31" src="https://github.com/user-attachments/assets/e88a60ee-962e-4a5f-b466-4e3e6a259a92">
 
 
<img width="1261" alt="Screenshot 2024-09-27 at 04 14 52" src="https://github.com/user-attachments/assets/75edea95-1217-483e-8bf4-80c302929094">
 
 
<img width="1322" alt="Screenshot 2024-09-27 at 04 15 17" src="https://github.com/user-attachments/assets/6df40f7f-0450-4866-a83e-d66bdedd7af1">
 
 
Round update:
 
 
<img width="1290" alt="Screenshot 2024-09-27 at 04 17 59" src="https://github.com/user-attachments/assets/8aeb9b6e-3ea1-47ee-a039-a9481df0f761">
 
 
<img width="1318" alt="Screenshot 2024-09-27 at 04 35 59" src="https://github.com/user-attachments/assets/ec3d0b77-e250-4f77-87b6-3a43e67df2a1">
 
 
<img width="1320" alt="Screenshot 2024-09-27 at 04 36 30" src="https://github.com/user-attachments/assets/3406b437-8cda-4ff0-8291-793c59670963">
 
 
<img width="1294" alt="Screenshot 2024-09-27 at 04 37 00" src="https://github.com/user-attachments/assets/09c92841-1774-4488-a9b1-57dada244861">
